packages/cactus-plugin-ledger-connector-besu/src/main/json/openapi.json

```diff
@@ -828,6 +828,18 @@
                 ],
                 "additionalProperties": false,
                 "properties": {
+                    "caseID": {
```

Comment:
this is tied to the proposed plugin. however, our plugin requires the involved connectors to emit events based on issued transactions; and those transactions have an optional caseID. In other words, I believe the case ID must be on the transaction schema of each connector (although optional)

Reply:
@petermetz please let me know what you think of this. If you request that no changes are done to the connectors and their data schema at all, then probably the best implementation choice is to create a modified connector plugin that extends the base connector. But in that case, we will keep this code separate from the main codebase due to refactoring effort.

Reply:
> 

@RafaelAPB Sorry for the slow response again! This goes back to the same thing I was talking about a little higher up in the comments talking about RxJS streams made with subjects/observables.
If the cc viz plugin is the one where these case IDs logically belong then you could create an interface there that contains it and consume instances of that interface through the RxJS piping. Again, I'm happy to show some code on how this is done (it's pretty simple once you see it written down IMO)

This way you avoid putting stuff in the connector's more generic data model but also avoid creating a fork of it (the modified version that you suggested) so that we still only have to maintain the single codebase for the connector but it just has one tiny extra feature which is mostly decoupled from the other features of it (so complexity does not increase substantially but neither does maintenance burden)